modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -31,6 +31,101 @@
 )
 
 
+def lazy_metadata_decorator(
+    apply_axis=None, inherit=False, axis_arg=-1, transpose=False
+):
+    """
+    Lazily propagate metadata for the ``PandasDataframe``.
+
+    This decorator first adds the minimum required reindexing operations
+    to each partition's queue of functions to be lazily applied for
+    each PandasDataframe in the arguments by applying the function
+    run_f_on_minimally_updated_metadata. The decorator also sets the
+    flags for deferred metadata synchronization on the function result
+    if necessary.
+
+    Parameters
+    ----------
+    apply_axis : str, default: None
+        The axes on which to apply the index object to the `self._partitions` lazily.
+        Case "both": Add reindexing operations on both axes to partition queue.
+        Case "opposite": Add reindexing operations for reduce functions dependending on axis argument to partition queue.
```

Comment:
For "opposite", I think we should clarify that we reindex the axis complementary to the `apply_axis`.